src/modules/form/hooks/usePreloadPicklists.tsx

```diff
@@ -0,0 +1,79 @@
+import { useLazyQuery } from '@apollo/client';
+import { compact, isEmpty } from 'lodash-es';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { ItemMap } from '../types';
+
+import useStabilizedValue from '@/hooks/useStabilizedValue';
+import {
+  GetPickListDocument,
+  PickListOptionFieldsFragment,
+  PickListType,
+} from '@/types/gqlTypes';
+
+const usePreloadPicklists = (
+  itemMap: ItemMap | undefined,
+  relationId?: string
+) => {
+  const [fetchPickList] = useLazyQuery(GetPickListDocument);
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState<PickListOptionFieldsFragment[]>([]);
+  // Stabilize the value in case we're not careful about reference equality on the input
+  const stabilizedItemMap = useStabilizedValue(itemMap);
+
+  const pickListTypesToFetch = useMemo(
+    () =>
+      compact(
+        Object.values(stabilizedItemMap || {})
+          .map((item) => (item.hidden ? null : item.pickListReference))
+          .filter((reference) => {
+            const isValid =
+              reference &&
+              Object.values<string>(PickListType).includes(reference);
+
+            if (isValid) return true;
+
+            return false;
+          })
+      ),
+    [stabilizedItemMap]
+  );
+
+  const fetch = useCallback(() => {
+    if (isEmpty(pickListTypesToFetch)) {
+      setLoading(false);
+      return;
+    }
+
+    setLoading(true);
+    Promise.all(
+      pickListTypesToFetch.map((pickListType) =>
+        fetchPickList({ variables: { relationId, pickListType } })
+      )
+    )
+      .then((results) => {
+        setData(
+          results.map(
+            (result) => result.data.pickList as PickListOptionFieldsFragment
+          )
+        );
+      })
+      .catch(() => setData([]))
+      .finally(() => setLoading(false));
+  }, [fetchPickList, pickListTypesToFetch, relationId]);
+
+  useEffect(() => {
+    fetch();
+  }, [fetch]);
```

Comment:
I want to expose the `fetch` function in the return value of the hook, which gives us a way to force-refetch the pick lists should we need to. In order to do that I need to declare it as a const using `useCallback`. Since I also want to call it in this hook on mount and whenever the deps change, I just call the `fetch` function inside of a `useEffect`.